src/django_components/slots.py

```diff
@@ -677,9 +685,9 @@ def render(self, context: Context, name: str, **kwargs: Any) -> SafeString:
         # In this case, we need to find the context that was used to render the component,
         # and use the fills from that context.
         if (
-            component_ctx.registry.settings.context_behavior == ContextBehavior.DJANGO
-            and component_ctx.outer_context is None
-            and (slot_name not in component_ctx.fills)
+            component.registry.settings.context_behavior == ContextBehavior.DJANGO
```

Comment:
One nice side effect of this change is that now inside the `{% slot %}` nodes we can access the `Component` instance that owns those slots.